src/elixirSenseAutocompleteProvider.ts

```diff
@@ -27,27 +27,23 @@ export class ElixirSenseAutocompleteProvider implements vscode.CompletionItemPro
                 return;
             }
 
-            if (!this.elixirSenseClient) {
-                console.log('ElixirSense client not ready');
-                console.error('rejecting');
-                reject();
-                return;
-            }
             const payload = {
                 buffer : document.getText(),
                 line   : position.line + 1,
                 column : position.character + 1
             };
 
```

Comment:
This is how you correctly use promises, btw :1st_place_medal: 

They're kinda like elixir's pipe chains, describing the happy path chained with `.then` and with the `.catch` handling whatever gets thrown. Combined with the validator pattern, the code practically reads like english. 
It's easy to modify, and super DRY.